Prevent accidental submission of Tavern/Guild posts after pasting

siege918 opened the pull request:
Fixes #10117

### Changes
Upon pasting into the post boxes on the Tavern or Guild pages, a boolean flag is set to temporarily disable the ctrl-enter shortcut for posting. Currently, this disables the shortcut for .5 seconds, but this value can be easily adjusted (on lines website/client/components/groups/group.vue:571 and website/client/components/groups/tavern.vue:764).

I had issues running tests locally (both on develop and my own branch), so I don't know how they behave until I see the results from submitting this PR.

----
UUID: 8b5e8110-0d52-43d5-8c20-998182d0b13c

Commit:
Temporarily disable ctrl-enter to send Guild messages after paste

Disable Ctrl-Enter after pasting, because some users are experiencing issues with accidentally sending their messages after pasting.

Commit:
Code style fixes for "Temporarily disable ctrl-enter to send Guild messages after paste"

negue commented on website/client/components/groups/group.vue:
```diff
@@ -552,6 +552,30 @@ export default {
+    disableMessageSendShortcut () {
+      // Some users were experiencing accidental sending of messages after pasting
+      // So, after pasting, disable the shortcut for a second.
+      this.chatSubmitDisable = true;
```
custom variables should go into `data()`

negue commented on website/client/components/groups/group.vue:
```diff
@@ -552,6 +552,30 @@ export default {
+      let This = this;
+
+      this.chatSubmitTimeout = window.setTimeout(() => {
+        This.chatSubmitTimeout = null;
```
there shoudn't be a need to use temp variables for `this` during compilation everything inside a `lamdba method` can access the outer scope

negue reviewed (changes requested):
the fix looks solid, could you try to extract the textarea from both `vue`s and have it inside its own component? that way we'd only have one component to fix in the future

siege918 commented:
I'll take a look at these this afternoon.

Commit:
Merge branch 'develop' of https://github.com/HabitRPG/habitica into siege918/disable-ctrlenter-after-paste

Commit:
Fix issues with variable location

siege918 commented:
Committed a couple of changes to address your comments, but going to take a little longer to see if I can abstract the textbox. This is admittedly my first time working in vue.js but I'll see what I can do.

Commit:
Fix variables for accidental chat submission features

Moving vatiables for the chat submit timeout to their own variable so they won't be overwritten

siege918 commented:
I've tried to extract the component, but it's clear that I don't know vue.js or Habitica well enough to make sure all the components and dependencies for guild chat are carried into the component correctly.

negue commented on website/client/components/groups/group.vue:
```diff
@@ -397,6 +397,10 @@ export default {
         challenges: true,
       },
+	  chat: {
+		SubmitDisable: false,
```
could you use the `camelCase` naming for the inner properties here?

siege918 replied:
My mistake! Missed this when I refactored. I'll get that fixed.

negue commented:
@siege918 don't worry, if you can't extract it I can do it in a different PR and ping you in that one (if you want to see the refactoring^^)

if you fix the lint issues then I guess its ready 👌

siege918 commented:
Definitely do ping me on the refactor, I'm interested to see!

Commit:
Fix code formatting issues with accidental chat submission code

Commit:
Remove leading space from variables to fix lint issues

siege918 commented:
@negue Should be in good shape now!

paglias reviewed (changes requested):
@siege918 thanks for this, the code looks good but it looks like it's duplicate between two files, can you consolidate the shared code in a single component / mixin? Thanks

siege918 commented:
Hi @paglias! As I mentioned further up, I don't feel comfortable doing so due to this being my first time using vue.js and looking at Habitica, and I couldn't wrap my head around the dependencies that I'd have to change to make it work.

@negue offered to make this change in a separate PR instead.

paglias commented:
Oh, sorry, I missed that part!

negue commented:
Sorry it took so long, LGTM 👍 and I'll ping you once this is merged for the extracted component

SabreCat commented:
Thanks for this, @siege918, and welcome to the ranks of the Blacksmiths at tier 1! This will undergo additional testing in a staging environment for a few days, and if all is well go live in production around the end of next week.

Note that achieving further tiers requires increasing amounts of effort from one to the next, so Tier 2 may not come immediately after your next PR. But keep helping out and we'll express our gratitude accordingly!
